packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -982,8 +982,13 @@ export default function<T, P, I, TI, HI, PI, C, CC, CX, PL>(
       // Subscribe to all changes by default
       observedBits = MAX_SIGNED_31_BIT_INT;
     }
+
     // Store the observedBits on the fiber's stateNode for quick access.
-    workInProgress.stateNode = observedBits;
+    if (oldProps === null) {
+      workInProgress.stateNode = {bits: observedBits};
+    } else {
+      workInProgress.stateNode.bits = observedBits;
```

Comment:
This will mutate both the current and the workInProgress since they'd share the same object. Generally, that's a big nono because many things can go wrong. We'd have to prove that this can't have those issues and won't get them later somehow.

Reply:
Been too long since I work on Fiber. Thanks for catching.